File: sorokac_u2/graticule.m
function [XM, YM, XP, YP] = graticule(umin, umax, vmin, vmax, Du, Dv, du, dv, R, uk, vk, u0, proj)
    %Create graticule
    
    %Create list of meridians
    XM = []; YM = [];
    for v = vmin:Dv:vmax
    
        %Create meridian
        um = umin:du:umax;
        n = length(um);
        vm = ones(1, n)*v;
        
        %Convert to oblique aspect
        [sm, dm] = uvTosd(um, vm, uk, vk);
    
        %Compute xm, ym
        [xm, ym] = proj(R, sm, dm, u0);
    
        %Add meridian to the list
        XM = [XM; xm];
        YM = [YM; ym];
    
    end
    
    %Create list of parallels
    XP = []; YP = [];
    for u = umin:Du:umax
        
        %Create parallel
        vp = vmin:dv:vmax;
        n = length(vp);
        up = ones(1, n)*u;
        
        %Convert to oblique aspect
        [sp, dp] = uvTosd(up, vp, uk, vk);
    
        %Compute xp, yp
        [xp, yp] = proj(R, sp, dp, u0);
    
        %Add parallel to the list
        XP = [XP; xp];
        YP = [YP; yp];
    
    end

end

File: uloha_3/graticule.py
from math import *
from uvTosd import uv_sd

def graticule(umin, umax, vmin, vmax, Du, Dv, du, dv, R, uk, vk, s0, rho0, proj):
    #Create list of meridians
    XM, YM = [], []
    
    v = vmin
    while v <= vmax:
        #Create meridian points
        xm_line, ym_line = [], []
        u = umin
        while u <= umax:
            #Convert to oblique aspect
            s, d = uv_sd(u, v, uk, vk)
            
            #Compute xm, ym using the passed projection function
            xm, ym = proj(R, s, d, s0, rho0)
            
            xm_line.append(xm)
            ym_line.append(ym)
            u += du
        
        #record the meridian    
        XM.append(xm_line)
        YM.append(ym_line)
        v += Dv

    #Create list of parallels
    XP, YP = [], []
    
    u = umin
    while u <= umax:
        # Create parallel points
        xp_line, yp_line = [], []
        v = vmin
        while v <= vmax:
            #Convert to oblique aspect
            s, d = uv_sd(u, v, uk, vk)
            
            #Compute xp, yp using the passed projection function
            xp, yp = proj(R, s, d, s0, rho0)
            
            xp_line.append(xp)
            yp_line.append(yp)
            v += dv
            
        #record the parallel    
        XP.append(xp_line)
        YP.append(yp_line)
        u += Du

    return XM, YM, XP, YP